src/com/goide/editor/GoFoldingBuilder.java

```diff
@@ -130,22 +130,60 @@ protected void buildLanguageFoldRegions(@NotNull final List<FoldingDescriptor> r
       foldTypes(type.getSpecType().getType(), result);
     }
 
+    for (GoExprCaseClause caseClause : PsiTreeUtil.findChildrenOfType(file, GoExprCaseClause.class)) {
+      if (caseClause.getColon() == null) continue;
+      TextRange range = processList(caseClause.getColon().getNextSibling(),
+                                    ContainerUtil.getLastItem(caseClause.getStatementList()),
+                                    caseClause.getStatementList().size(), 1);
+      if (range != null) {
+        result.add(new FoldingDescriptor(caseClause, range));
+      }
+    }
+
+    for (GoExprSwitchStatement switchStatement : PsiTreeUtil.findChildrenOfType(file, GoExprSwitchStatement.class)) {
+      TextRange range = processList(switchStatement.getLbrace(),
+                                    switchStatement.getRbrace(),
+                                    switchStatement.getExprCaseClauseList().size(), 1);
+      if (range != null) {
+        result.add(new FoldingDescriptor(switchStatement, range));
+      }
+    }
+
+    for (GoCommClause commClause : PsiTreeUtil.findChildrenOfType(file, GoCommClause.class)) {
+      if (commClause.getColon() == null) continue;
+      TextRange range = processList(commClause.getColon().getNextSibling(),
+                                    ContainerUtil.getLastItem(commClause.getStatementList()),
```

Comment:
I asked not to use this logic only for `{...}` blocks

`getNextSiblign` looks not good here